modules/samtools/collate/main.nf

```diff
@@ -0,0 +1,36 @@
+process SAMTOOLS_COLLATE {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::samtools=1.15.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.15.1--h1170115_0':
+        'quay.io/biocontainers/samtools:1.15.1--h1170115_0' }"
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
```

Comment:
`samtools collate` can also output in SAM, BAM or CRAM format (see `--output-fmt`). You can specify what the output format is in the `args` (example: https://github.com/nf-core/modules/blob/d5d785b3d8b422cda9c6d84a23f629a8e9ff8cd8/modules/bcftools/convert/main.nf#L31-L35)